test/t/module.rb

```diff
@@ -275,6 +275,16 @@ def self.const_missing(sym)
   assert_equal 42, Test4ConstMissing.const_get(:ConstDoesntExist)
 end
 
+assert('Module#extend_object', '15.2.2.4.25') do
+  cls = Class.new
+  mod = Module.new { def foo; end }
+  a = cls.new
+  b = cls.new
+  assert_same a, mod.extend_object(a)
```

Comment:
But the return value from `append_features`, `extend_object` is implementation-defined. So I don't think we need to test the return value here.

Reply:
The specifications of `Module#extend_object` in ISO (final draft) is as follows:

> Let S be the singleton class of object. Invoke the method append features (see
5 15.2.2.4.10) on the receiver with S as the only argument, and return the resulting value.

I think this means returning the object of the argument. Is my interpretation wrong or has it been changed in the final version?

Reply:
It reads "`extend_object` returns the return value from `append_feature`" and the return value from `append_feature` is implementation defined.

Reply:
Thank you for your elaboration.

Then current mruby conform to ISO, but CRuby does not conform to ISO because it does not return the return value from `append_features`.

Should mruby conform to ISO or CRuby?

Reply:
ISO here, I think. If the return value is meaningful, it should be compatible with CRuby. But this case is not.

Reply:
Then, there is no need to change it, so I will close this PR.